yt/frontends/stream/data_structures.py

```diff
@@ -439,9 +443,13 @@ def assign_particle_data(ds, pdata, bbox):
             grid = {"number_of_particles": 0}
             grid_pdata.append(grid)
 
+        particle_index_fields = [
```

Comment:
looks to me that this repetition isn't necessary. You could define `particle_index_fields` as a property
```python
@property
def particle_index_fields(self):
    return [f"particle_position_{ax}" for ax in ds.coordinates.axis_order] + ["particle_position"]

```
and then reuse this downstream

Reply:
So, yes, I agree... sort of.

What I'm nervous about here is the idea that we're edging closer and closer to refactoring into having index fields listed, which is a *good* thing, but which I would like to do much more deliberately.

Reply:
that's a valid point. Unfortunately I lack a vision of the big picture here so I'm focusing on the scale of this single PR instead. I agree that it'd be much better to start from the ground up if we're going to refactor this at a lower level.